### PR TITLE
Consolidate Vega Chart Renderer

### PR DESCRIPTION
**Pull request by briangregoryholmes**
This consolidates the functionality of the two VegaRenderer components used between dashboards and charts and adds a helper sizing element to ensure that the chart takes up the full width and height of its container.

### CODE REVIEW

File: web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte [hunk 1]
```diff
@@ -287,7 +287,10 @@
           ? $isMeasureValidPercentOfTotal(measure.name)
           : false}
 
-        <div class="flex flex-row gap-x-7">
+        <div
+          class="flex flex-row gap-x-7"
+          style:height="{expandedMeasureName ? 220 : 128}px"
+        >
```
Comment (djbarnwal) [inline web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte:293]: Is this "220" from figma mocks? TDD charts look smaller in height than before

File: web-common/src/features/charts/ChartDisplay.svelte [hunk 2]
```diff
@@ -47,7 +47,9 @@
     {:else if !parsedVegaSpec}
       <p>Chart not available</p>
     {:else}
-      <VegaLiteRenderer {data} spec={parsedVegaSpec} />
+      <div class="w-full h-1/2">
```
[the remark this answers is not in the document]
Comment (briangregoryholmes) [reply web-common/src/features/charts/ChartDisplay.svelte:50]: Not sure we'll be able to resolve this for the initial release, but I'll continue to dig into the best option here. Setting width to container isn't valid for specs that have repeating graphs.

File: web-common/src/features/charts/render/VegaLiteRenderer.svelte [hunk 3]
```diff
@@ -1,43 +1,61 @@
...
+  export let spec: VisualizationSpec;
   export let signalListeners: SignalListeners = {};
+  export let error: string | null = null;
 
-  let options: EmbedOptions = {
+  let contentRect = new DOMRect(0, 0, 0, 0);
+  let viewVL: View;
+
+  $: width = contentRect.width;
+  $: height = contentRect.height - 30;
```
[the remark this answers is not in the document]
Comment (briangregoryholmes) [reply web-common/src/features/charts/render/VegaLiteRenderer.svelte:21]: Unfortunately, no. What I've decided to do is just update the charts page to have a version of the full custom dashboard view, which I think is the right call.

File: web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte (hunk 1 above)
Comment (briangregoryholmes) [reply web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte:293]: Will update

File: web-common/src/features/charts/ChartDisplay.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (djbarnwal) [reply web-common/src/features/charts/ChartDisplay.svelte]: Even for simpler charts we might get into layout issues - 
image

[the remark this answers is not in the document]
Comment (briangregoryholmes) [reply web-common/src/features/charts/ChartDisplay.svelte]: Looks like Vega overrides fit-y with certain specs. Adjusted to address this.

**Review by djbarnwal — APPROVED** [review]

**Pull request merged**